### PR TITLE
fix(ctb): Bond payouts

### PR DESCRIPTION
**Pull request by clabby**
## Overview

Adds a fix for two issues with the bond draft, where the party who resolved was being paid out, and the final subgame did not pay out to the party who stepped against it.

**Metadata**
closes https://github.com/ethereum-optimism/client-pod/issues/504
closes https://github.com/ethereum-optimism/client-pod/issues/505

### CODE REVIEW

**Comment by clabby** [comment]
Current dependencies on/for this PR:
* `develop`
  * **PR #9258** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/9258?utm_source=stack-comment-icon" target="_blank">Graphite</a> 
    * **PR #9259** <a href="https://app.graphite.dev/github/pr/ethereum-optimism/optimism/9259?utm_source=stack-comment-icon" target="_blank">Graphite</a>  👈

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/ethereum-optimism/optimism/9259?utm_source=stack-comment).

**Review by Inphi — APPROVED** [review]

File: packages/contracts-bedrock/src/dispute/FaultDisputeGame.sol [hunk 1]
```diff
@@ -389,7 +389,13 @@ contract FaultDisputeGame is IFaultDisputeGame, Clone, ISemver {
...
         // and return early.
         if (challengeIndicesLen == 0 && _claimIndex != 0) {
-            _distributeBond(parent.claimant, parent);
+            // In the event that the parent claim is at the max depth, there will always be 0 subgames. If the
+            // `counteredBy` field is set and there are no subgames, this implies that the parent claim was successfully
+            // stepped against. In this case, we pay out the bond to the party that stepped against the parent claim.
+            // Otherwise, the parent claim is uncontested, and the bond is returned to the claimant.
+            address counteredBy = parent.counteredBy;
+            address recipient = counteredBy == address(0) ? parent.claimant : counteredBy;
+            _distributeBond(recipient, parent);
```
Comment (refcell) [inline packages/contracts-bedrock/src/dispute/FaultDisputeGame.sol:398]: Shouldn't the challenger who made the claim at max depth be the one to receive the bond as opposed to the actor who called step?

Comment (clabby) [reply packages/contracts-bedrock/src/dispute/FaultDisputeGame.sol:398]: Not necessarily - if the claim at max depth is countered, the person who disproved it with a step receives their bond. If it's uncountered, they get their bond back.

Comment (refcell) [reply packages/contracts-bedrock/src/dispute/FaultDisputeGame.sol:398]: Makes sense, sweet!

**Comment by Inphi** [comment]
I think this needs a rebase to fix codecov.

**Comment by refcell** [comment]
> I think this needs a rebase to fix codecov.

Codecov doesn't block merging, it's just a report :)

**Pull request merged**